microprofile/fault-tolerance/src/main/java/io/helidon/microprofile/faulttolerance/MethodIntrospector.java

```diff
@@ -65,7 +65,7 @@ class MethodIntrospector {
                         .stream()
                         .filter(am -> am.getJavaMember().equals(method))
                         .findFirst();
-        this.annotatedMethod = annotatedMethodOptional.orElseThrow();
+        this.annotatedMethod = annotatedMethodOptional.orElse(new FtAnnotatedMethod(method));
```

Comment:
Is the `AnnotatedType` derived from the `beanClass` valid for the `FtAnnotatedMethod` case as well? If so, would it make sense to pass that into the `FtAnnotatedMethod` constructor instead of having the constructor creating its own impl of `AnnotatedType` for the wrapper method with several "should not be called" methods?

Reply:
Good question, no in the proxy case (something new to me) the type of the intercepted `beanClass` is `org.glassfish.jersey.microprofile.restclient.MethodModel` and that obviously has no annotations for us to directly inspect. Hence, the annotated type is derived from the Java method's declaring type. I believe digging into `MethodModel` we could find the real type, but that's deep into Jersey's implementation, which isn't ideal.